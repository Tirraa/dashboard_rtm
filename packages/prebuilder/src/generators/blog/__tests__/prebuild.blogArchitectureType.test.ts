// eslint-disable-next-line import/no-extraneous-dependencies
import { describe, expect, it } from 'vitest';

import generateBlogArchitectureType from '../blogArchitectureType';

const fs = require('fs/promises');

const __TARGET_FOLDER_ROOT = './packages/prebuilder/src/generators/blog/__tests__/FAKE_CODEGEN';
const __TARGET_FOLDER = __TARGET_FOLDER_ROOT + '/' + 'BLOG_ARCHITECTURE_TYPE';

describe('generateBlogArchitectureType', () => {
  it('should match snapshot', async () => {
    const targetFile = 'FAKE_EMPTY_BLOG_ARCHITECTURE_TYPE';
    await generateBlogArchitectureType({}, targetFile, __TARGET_FOLDER);

    const fileContent = await fs.readFile(`${__TARGET_FOLDER}/${targetFile}.ts`, 'utf8');
    expect(fileContent).toMatchSnapshot();
  });

  it('should match snapshot', async () => {
    const targetFile = 'FAKE_BLOG_ARCHITECTURE_TYPE';
    await generateBlogArchitectureType(
      {
        'fake-category-one': {
          'fake-subcategory-one': {
            DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'],
            en: ['fake-post-01', 'fake-post-03'],
            it: ['fake-post-01', 'fake-post-02'],
            fr: ['fake-post-01']
          },
          'fake-subcategory-three': {
            DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'],
            es: ['fake-post-01'],
            zh: ['fake-post-02']
          },
          'fake-subcategory-two': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], fr: ['fake-post-02'] }
        },
        'fake-category-three': {
          'fake-subcategory-five': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], en: ['fake-post-03'], it: ['fake-post-03'] },
          'fake-subcategory-four': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], es: ['fake-post-04'] }
        },
        'fake-category-two': { 'fake-subcategory-six': { DEFAULT_LANGUAGE: ['fake-post-01', 'fake-post-02', 'fake-post-03'], zh: ['fake-post-10'] } }
      },
      targetFile,
      __TARGET_FOLDER
    );

    const fileContent = await fs.readFile(`${__TARGET_FOLDER}/${targetFile}.ts`, 'utf8');
    expect(fileContent).toMatchSnapshot();
  });
});
