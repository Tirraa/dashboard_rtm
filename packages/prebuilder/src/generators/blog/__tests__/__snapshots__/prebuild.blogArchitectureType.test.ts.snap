// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`generateBlogArchitectureType > should match snapshot 1`] = `
"// AUTOGENERATED: Don't edit this file!
type FAKE_EMPTY_BLOG_ARCHITECTURE_TYPE = {};
export default FAKE_EMPTY_BLOG_ARCHITECTURE_TYPE;
"
`;

exports[`generateBlogArchitectureType > should match snapshot 2`] = `
"// AUTOGENERATED: Don't edit this file!
type FAKE_BLOG_ARCHITECTURE_TYPE = {
  'fake-category-one': 'fake-subcategory-one' | 'fake-subcategory-three' | 'fake-subcategory-two';
  'fake-category-three': 'fake-subcategory-five' | 'fake-subcategory-four';
  'fake-category-two': 'fake-subcategory-six';
};
export default FAKE_BLOG_ARCHITECTURE_TYPE;
"
`;
